### PR TITLE
@wdio/globals: use declare var

### LINKED ISSUE #9280: [🐛 Bug]: `webdriverio` v8 seems to depend on `@wdio/globals`
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

8.0.0

### Node.js Version

16

### Mode

WDIO Testrunner

### Which capabilities are you using?

```typescript
N/A
```

### What happened?

`@serenity-js/webdriverio` module depends on the `webdriverio` module.
However, it seems like in version 8, `webdriverio` depends on the `WebdriverIO` namespace from `@wdio/globals`

See:
- https://github.com/webdriverio/webdriverio/blob/main/packages/webdriverio/src/types.ts#L37
- https://github.com/webdriverio/webdriverio/blob/main/packages/wdio-globals/types.d.ts#L14

### What is your expected behavior?

It would be ideal if `webdriverio` did not depend on `@wdio/globals` and defined its own `WebdriverIO` namespace.

### How to reproduce the bug.

- remove dependency on `@types/wdio/async` (from WDIO 7)
- don't add the dependency on `@wdio/globals/types` (to avoid polluting the global namespace with `expect`)
- run WebdriverIO while importing types like `browser` or `$` directly from the `webdriverio` module

### Relevant log output

```typescript
@serenity-js/webdriverio: node_modules/webdriverio/build/commands/browser/$$.d.ts:58:46 - error TS2694: Namespace 'global.WebdriverIO' has no exported member 'Browser'.
@serenity-js/webdriverio: 58 export default function $$(this: WebdriverIO.Browser | WebdriverIO.Element, selector: Selector | ElementReference[] | WebdriverIO.Element[]): Promise<ElementArray>;
@serenity-js/webdriverio:                                                 ~~~~~~~
@serenity-js/webdriverio: node_modules/webdriverio/build/commands/browser/$$.d.ts:58:68 - error TS2694: Namespace 'global.WebdriverIO' has no exported member 'Element'.
@serenity-js/webdriverio: 58 export default function $$(this: WebdriverIO.Browser | WebdriverIO.Element, selector: Selector | ElementReference[] | WebdriverIO.Element[]): Promise<ElementArray>;
@serenity-js/webdriverio:                                                                       ~~~~~~~
@serenity-js/webdriverio: node_modules/webdriverio/build/commands/browser/$$.d.ts:58:131 - error TS2694: Namespace 'global.WebdriverIO' has no exported member 'Element'.
@serenity-js/webdriverio: 58 export default function $$(this: WebdriverIO.Browser | WebdriverIO.Element, selector: Selector | ElementReference[] | WebdriverIO.Element[]): Promise<ElementArray>;
@serenity-js/webdriverio:                                                                                                                                      ~~~~~~~
@serenity-js/webdriverio: node_modules/webdriverio/build/commands/browser/$.d.ts:76:45 - error TS2694: Namespace 'global.WebdriverIO' has no exported member 'Browser'.
@serenity-js/webdriverio: 76 export default function $(this: WebdriverIO.Browser | WebdriverIO.Element, selector: Selector): Promise<WebdriverIO.Element>;
@serenity-js/webdriverio:                                                ~~~~~~~
@serenity-js/webdriverio: node_modules/webdriverio/build/commands/browser/$.d.ts:76:67 - error TS2694: Namespace 'global.WebdriverIO' has no exported member 'Element'.
@serenity-js/webdriverio: 76 export default function $(this: WebdriverIO.Browser | WebdriverIO.Element, selector: Selector): Promise<WebdriverIO.Element>;
@serenity-js/webdriverio:                                                                      ~~~~~~~
@serenity-js/webdriverio: node_modules/webdriverio/build/commands/browser/$.d.ts:76:117 - error TS2694: Namespace 'global.WebdriverIO' has no exported member 'Element'.
@serenity-js/webdriverio: 76 export default function $(this: WebdriverIO.Browser | WebdriverIO.Element, selector: Selector): Promise<WebdriverIO.Element>;

```
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### PR DESCRIPTION
**Pull request by SCG82**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

Resolves #9280 

- use `declare var` to fix https://github.com/webdriverio/webdriverio/blob/main/packages/wdio-cli/src/commands/repl.ts#L55
- remove expect declaration to avoid clashing with expect-webdriverio
- expose /types (node16 module resolution)
- @wdio/runner: fix type errors

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Comment by christian-bromann** [comment]
Might this be a fix for #9280?

**Comment by SCG82** [comment]
> Might this be a fix for #9280?

That's the issue I'm trying to fix with this.

**Comment by SCG82** [comment]
Might need to replace all uses of `WebdriverIO.XXX` with explicit imports for all packages.

**Comment by christian-bromann** [comment]
See #9256 .. I am totally open to reconsider how everything is typed as long as we don't break things. If I apply your fix to an example project I still get:

```
const expect: ExpectWebdriverIO.Expect
Cannot redeclare block-scoped variable 'expect'.ts(2451)
jest-global.d.ts(3, 15): 'expect' was also declared here.
```

You think there are other changes needed for this in `expect-webdriverio`?

**Comment by SCG82** [comment]
@christian-bromann Can you take a look now?

File: packages/wdio-globals/types.d.ts [hunk 1]
```diff
@@ -33,4 +33,4 @@ declare function $$(...args: Parameters<WebdriverIO.Browser['$$']>): ReturnType<
 declare var browser: WebdriverIO.Browser
 declare var driver: WebdriverIO.Browser
 declare var multiremotebrowser: WebdriverIO.MultiRemoteBrowser
-// declare var expect: Expect
+// declare var expect: ExpectType
```
Comment (christian-bromann) [inline packages/wdio-globals/types.d.ts:36]: Can we enable or remove this?

**Review by christian-bromann — COMMENTED** [review]
one comment, otherwise this looks good to me

**Comment by christian-bromann** [comment]
> expose /types (node16 module resolution)
> @wdio/runner: fix type errors

I don't see these things in the PR

File: packages/wdio-globals/types.d.ts (hunk 1 above)
Comment (SCG82) [reply packages/wdio-globals/types.d.ts]: done

File: packages/wdio-runner/src/browser.ts [hunk 2]
```diff
@@ -12,9 +12,9 @@ import type { TestFramework } from './types'
 const log = logger('@wdio/runner')
 const sep = '\n  - '
 
-type WDIOErrorEvent = Pick<ErrorEvent, 'filename' | 'message'> & { hasViteError?: boolean }
+type WDIOErrorEvent = Partial<Pick<ErrorEvent, 'filename' | 'message'>> & { hasViteError?: boolean }
 interface TestState {
-    failures?: number
+    failures?: number | null
```
Comment (christian-bromann) [inline packages/wdio-runner/src/browser.ts:17]: Is this change related to the PR/original problem?

Comment (SCG82) [reply packages/wdio-runner/src/browser.ts:17]: No, to fix errors that appeared when running setup-full after rebasing.

**Review by christian-bromann — APPROVED** [review]
👍  LGTM

**Comment by christian-bromann** [comment]
Thanks a ton!

**Pull request merged**